### ISSUE TITLE
[inductor] `le` view is different between Inductor and Torch

### ISSUE THREAD

**Issue by anijain2305** [pytorch/torchdynamo#1564]
I am not sure if this is a problem or really a root cause of the problems I am seeing. But, wanted to get other opinions on this.

@ngimel @jansel 

~~~

import torch
from torch import tensor, device
import torch.fx as fx
from torchdynamo.testing import rand_strided
from math import inf
from torch.fx.experimental.proxy_tensor import make_fx

# torch version: 1.14.0a0+git4bfe2a2
# torch cuda version: 11.6
# torch git version: 4bfe2a24505049fa4fe43d24c2e3a5f5d99d9f00


# CUDA Info:
# nvcc: NVIDIA (R) Cuda compiler driver
# Copyright (c) 2005-2022 NVIDIA Corporation
# Built on Thu_Feb_10_18:23:41_PST_2022
# Cuda compilation tools, release 11.6, V11.6.112
# Build cuda_11.6.r11.6/compiler.30978841_0

# GPU Hardware Info:
# NVIDIA A100-SXM4-40GB : 8


from torch.nn import *
class Repro(torch.nn.Module):
    def __init__(self):
        super().__init__()



    def forward(self, _to_copy_402):
        le_23 = torch.ops.aten.le.Scalar(_to_copy_402, 0);  _to_copy_402 = None
        return (le_23,)

args = [((2, 192, 17, 17), (55488, 289, 17, 1), torch.float32, 'cuda')]
args = [rand_strided(sh, st, dt, dev) for (sh, st, dt, dev) in args]
mod = make_fx(Repro())(*args)


ref = mod(*args)[0]

from torchinductor.compile_fx import compile_fx_inner
from torchdynamo.debug_utils import same_two_models

compiled = compile_fx_inner(mod, args)
res = compiled(*args)[0]

assert torch.allclose(ref, res)

ref_u8 = ref.view(dtype=torch.uint8)
res_u8 = res.view(dtype=torch.uint8)
assert torch.allclose(ref_u8, res_u8)
~~~

**Comment by jansel** [comment]
It appears that `True` in Triton is `0xFF`, while `True` in eager is `0x01`

**Comment by jansel** [comment]
I was expecting this patch to fix it:
```patch
diff --git a/torchinductor/codegen/triton.py b/torchinductor/codegen/triton.py
index 3ce58b7..da9c09d 100644
--- a/torchinductor/codegen/triton.py
+++ b/torchinductor/codegen/triton.py
@@ -787,6 +787,8 @@ class TritonKernel(Kernel):
     def store(self, name, index, value, mode=None):
         var = self.args.output(name)
         index, mask = self.indexing(index, value, dense_indexing=True)
+        if V.graph.get_dtype(name) is torch.bool:
+            value = f"tl.where({value}, 1, 0)"
         if mode is None:
             line = f"tl.store({var} + ({index}), {value}, {mask})"
         elif mode == "atomic_add":
```

However, it does *not*.  There must be some upstream logic in Triton that converts the `1` to `0xff` when you write to a bool tensor.  The easiest fix is likely on the Triton side, though we could just convert all bool tensors to int8 tensors inside inductor.

**Comment by jansel** [comment]
I filed an issue https://github.com/openai/triton/issues/756

**Comment by ngimel** [comment]
cc @desertfire, hopefully https://github.com/openai/triton/pull/746 should fix it.

**Comment by ngimel** [comment]
Note that pytorch eager converts non-standard bools to 1 on reads (there might be a few stray ops that don't do it, but @peterbell10 fixed most), and #1490 fixes non-standard bools in inductor-generated kernel reads, so by itself hopefully the issue above should not be observable, but ideally we should still have a fix like  https://github.com/openai/triton/pull/746 to avoid these workarounds.

**Comment by desertfire** [comment]
Verified https://github.com/pytorch/torchdynamo/pull/1585/ can fix this
